tls/s2n_security_policies.c

```diff
@@ -1224,6 +1228,13 @@ int s2n_connection_set_cipher_preferences(struct s2n_connection *conn, const cha
     /* If the security policy's minimum version is higher than what libcrypto supports, return an error. */
     POSIX_ENSURE((security_policy->minimum_protocol_version <= s2n_get_highest_fully_supported_tls_version()), S2N_ERR_PROTOCOL_VERSION_UNSUPPORTED);
 
+    /* If the certificates loaded in the config are incompatible with the security 
+     * policy's certificate preferences, return an error. */
+    if (conn->config != NULL) {
```

Comment:
nit: I think we can POSIX_ENSURE_REF the config here. I don't think the config can/should ever be null. If a config isn't set on a connection manually, it's set to the default config.